### PR TITLE
[#153] Implement Gif Resource Loading

### LINKED ISSUE #153: Investigate parsing GIFs to ex Animations
There are a lot of animation assets that would nice to use natively in ex without unpacking the gif into a spritesheet.

### PR DESCRIPTION
**Pull request by eonarheim**
See https://github.com/excaliburjs/Excalibur/pull/1075

@kevin192291's PR

Closes #153

### CODE REVIEW

**Comment by eonarheim** [comment]
@kevin192291 I did a few tweaks for docs and to remove the `_applyEffects()`, again I apologize for the confusion

**Comment by eonarheim** [comment]
@kevin192291 Let me know if you'd like to reset your fork's master back to the original and we can reopen your PR. Otherwise I'll merge this one instead.

**Comment by eonarheim** [comment]
If you have a local copy of your master still `git push -f` should restore your fork back to normal.

**Comment by kevin192291** [comment]
This looks good to me :) I don't think there is a need to re-open

File: src/spec/GifSpec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    gif.load().then(() => {
+      expect(gif).toBeDefined();
+      const spriteSheet: ex.SpriteSheet = gif.asSpriteSheet();
+      const sprite = spriteSheet.getSprite(0);
+      sprite.draw(engine.ctx, 0, 0);
+      imagediff.expectCanvasImageMatches('GifSpec/frame1.png', engine.canvas, done);
+    });
+  });
+
+  // it('should be read as an Animation', (done) => {
```
Comment (kamranayub) [inline src/spec/GifSpec.ts]: Reason for commenting out?

Comment (kevin192291) [reply src/spec/GifSpec.ts]: Ahh, I wasn't sure if it was important. Do we add tests later as well? or should all the tests be created up front?

Comment (eonarheim) [reply src/spec/GifSpec.ts]: @kevin192291 Tests are good, when they are added doesn't matter as much to me. 

We can just delete this if the test is not needed.

Comment (eonarheim) [reply src/spec/GifSpec.ts]: @kevin192291 I'd be open to merging this PR, and add a follow-up issue to address any @kamranayub's concerns

**Comment by eonarheim** [comment]
@kevin192291 Thanks again for helping out!

**Pull request merged**